### ISSUE TITLE
Invalid opType: undefined

### ISSUE THREAD

**Issue by sim1camporeale** [aws-amplify/amplify-js#5992]
Hi all, 
in this point there is some cases where the returned opType from merge function will be undefined. If the function put undefined value inside the opType array then the following switch case returns Invalid opType error. If think that you should handle the case when opType is undefined.

https://github.com/aws-amplify/amplify-js/blob/da2726d029c63d7472a32deffd1431322ec628ad/packages/datastore/src/sync/index.ts#L486

**Comment by manueliglesias** [comment]
Hi @sim1camporeale 

That `opType` comes from one of the storage adapters. Are you using DataStore in a browser or on react native?

Also, could you share the schema you are using?

**Comment by manueliglesias** [comment]
@sim1camporeale I am also curious, under what circumstances do you get undefined back from the merge function?

**Comment by sim1camporeale** [comment]
Hi manueliglesias,
I'm using DataStore in react-native with expo.
My schema is slighty complex if you want I can show it on discord.
The object that causes issues was this EdgeMetaData:

```
type EdgeMetaData
  @model
  @auth(
    rules: [
      {
        allow: groups
        groupsField: "users"
        groupClaim: "cognito:username"
        operations: [create, update, delete, read]
      }
      { allow: private, provider: iam, operations: [create, update, delete, read] }
    ]
  ) {
  id: ID!
  users: [String]!

  plants: [String]
  config: String
}

type Environment
  @model
  @auth(
    rules: [
      {
        allow: groups
        groupsField: "users"
        groupClaim: "cognito:username"
        operations: [create, update, delete, read]
      }
      { allow: private, provider: iam, operations: [create, update, delete, read] }
    ]
  ) {
  id: ID!
  places: [Place] @connection(keyName: "byEnv", fields: ["id"])
  schedules: [Schedule] @connection(keyName: "byEnv", fields: ["id"])
  points: [Point] @connection(keyName: "byEnv", fields: ["id"])
  edgeMetaDataId: ID!
  edgeMetaData: EdgeMetaData @connection(fields: ["edgeMetaDataId"])

  users: [String]!

  name: String
  address: String
}

```
It happens after some requests sometime. It happens more often when we write multiple times EdgeMetaData .

**Comment by manueliglesias** [comment]
Yesterday afternoon we published new versions, can you try with the latest?

**Comment by sim1camporeale** [comment]
Hi @manueliglesias  that issue was solved. But now we have an issue starting DataStore.
At first launch after we authenticate we wait for DataStore sync event using HUB. When HUB says that DataStore is ready we try to do a query against DataStore and it is empty. But we see subscription. If we close and reopen app we see data and we will be able to perform query.
We tried adding a sleep of 10 seconds after HUB event but nothing changes. I was looking for support on discord before submitting a new issue.
